### ISSUE TITLE
[VS 2017; latest build] Installation of extension fails

### ISSUE THREAD

**Issue by denis-golubev** [PistonDevelopers/VisualRust#280]
### Environment:
Windows 10 Pro N x64 (before Creators update)
Version: 1607
OS Build: 14393.1066

Visual Studio:
```
Microsoft Visual Studio Community 2017 
Version 15.1 (26403.7) Release
VisualStudio.15.Release/15.1.0+26403.7
Microsoft .NET Framework
Version 4.6.01586

Installed Version: Community

Visual Basic 2017   00369-60000-00001-AA658
Microsoft Visual Basic 2017

Visual C# 2017   00369-60000-00001-AA658
Microsoft Visual C# 2017

Visual C++ 2017   00369-60000-00001-AA658
Microsoft Visual C++ 2017

ASP.NET and Web Tools 2017   15.0.30320.0
ASP.NET and Web Tools 2017

GitHub.VisualStudio   2.2.0.10
A Visual Studio Extension that brings the GitHub Flow into Visual Studio.

JavaScript Language Service   2.0
JavaScript Language Service

JetBrains ReSharper Ultimate 2017.1.2    Build 108.0.20170428.75743
JetBrains ReSharper Ultimate package for Microsoft Visual Studio. For more information about ReSharper Ultimate, visit http://www.jetbrains.com/resharper. Copyright © 2017 JetBrains, Inc.

KofePackagePackage Extension   1.0
KofePackagePackage Visual Studio Extension Detailed Info

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Visual C++ for Linux Development   1.0.6
Visual C++ for Linux Development

Visual Studio tools for CMake   1.0
Visual Studio tools for CMake

Visual Studio Tools for Unity   3.1.0.0
Visual Studio Tools for Unity
```

### Steps to reproduce
1. Download build from: https://ci.appveyor.com/project/vosen/visualrust/build/152/artifacts
2. Perform installation

### Actual result
The installation fails with `The InstalledByMSI element in extension Visual Rust cannot be 'true' when installing an extension through the Extensions and Updates Installer.  The element can only be 'true' when an MSI lays down the extension manifest file.`. 

### Full log
```
12.05.2017 19:20:30 - Microsoft VSIX Installer
12.05.2017 19:20:30 - -------------------------------------------
12.05.2017 19:20:30 - vsixinstaller.exe version:
12.05.2017 19:20:30 - 15.0.26209.1 built by: D15REL
12.05.2017 19:20:30 - -------------------------------------------
12.05.2017 19:20:30 - Command line parameters:
12.05.2017 19:20:30 - C:\Program Files (x86)\Microsoft Visual Studio\Installer\resources\app\ServiceHub\Services\Microsoft.VisualStudio.Setup.Service\VSIXInstaller.exe,/admin,/logFile:C:\Users\XXXXXX\AppData\Local\Temp\Visual_Rust_for_Visual_Studio_2017_20170512192020_002_VSIXBootstrapper.log,C:\ProgramData\Package Cache\9864EE5369359DB622D0234B6C6B7640B949140C\VisualRust.vsix
12.05.2017 19:20:30 - -------------------------------------------
12.05.2017 19:20:30 - Microsoft VSIX Installer
12.05.2017 19:20:30 - -------------------------------------------
12.05.2017 19:20:31 - Initializing Install...
12.05.2017 19:20:32 - Extension Details...
12.05.2017 19:20:32 - 	Identifier         : 40c1d2b5-528b-4966-a7b1-1974e3568abe
12.05.2017 19:20:32 - 	Name               : Visual Rust
12.05.2017 19:20:32 - 	Author             : The Piston Project
12.05.2017 19:20:32 - 	Version            : 0.2.0
12.05.2017 19:20:32 - 	Description        : Visual Studio integration for the Rust programming language (http://www.rust-lang.org/)
12.05.2017 19:20:32 - 	Locale             : en-US
12.05.2017 19:20:32 - 	MoreInfoURL        : 
12.05.2017 19:20:32 - 	InstalledByMSI     : True
12.05.2017 19:20:32 - 	SupportedFrameworkVersionRange : [4.6,)
12.05.2017 19:20:32 - 
12.05.2017 19:20:32 - 	SignatureState     : Unsigned
12.05.2017 19:20:32 - 	Supported Products : 
12.05.2017 19:20:32 - 		Microsoft.VisualStudio.Community
12.05.2017 19:20:32 - 			Version : [15.0,)
12.05.2017 19:20:32 - 		Microsoft.VisualStudio.Pro
12.05.2017 19:20:32 - 			Version : [15.0,)
12.05.2017 19:20:32 - 		Microsoft.VisualStudio.Enterprise
12.05.2017 19:20:32 - 			Version : [15.0,)
12.05.2017 19:20:32 - 
12.05.2017 19:20:32 - 	References         : 
12.05.2017 19:20:32 - 	Prerequisites      : 
12.05.2017 19:20:32 - 		-------------------------------------------------------
12.05.2017 19:20:32 - 		Identifier   : Microsoft.VisualStudio.Component.CoreEditor
12.05.2017 19:20:32 - 		Name         : Visual Studio core editor
12.05.2017 19:20:32 - 		Version      : [15.0,)
12.05.2017 19:20:32 - 
12.05.2017 19:20:32 - 		-------------------------------------------------------
12.05.2017 19:20:32 - 		Identifier   : Microsoft.VisualStudio.Component.VC.CoreIde
12.05.2017 19:20:32 - 		Name         : Visual Studio C++ core features
12.05.2017 19:20:32 - 		Version      : [15.0,16.0)
12.05.2017 19:20:32 - 
12.05.2017 19:20:32 - Signature Details...
12.05.2017 19:20:32 - 	Extension is not signed.
12.05.2017 19:20:32 - 
12.05.2017 19:20:32 - Searching for applicable products...
12.05.2017 19:20:32 - Found installed product - Microsoft Visual Studio Community 2015
12.05.2017 19:20:32 - Found installed product - Microsoft Visual Studio 2015 Shell (Integrated)
12.05.2017 19:20:32 - Found installed product - Global Location
12.05.2017 19:20:32 - Found installed product - Visual Studio Community 2017
12.05.2017 19:20:44 - PKGDEF Information: Looking for master PkgDef file, Path: C:\Program Files (x86)\Visual Studio\2017\Community\Common7\IDE\master.pkgdef

12.05.2017 19:20:44 - PKGDEF Information: Creating PkgDefCacheNonVolatile
12.05.2017 19:20:44 - PKGDEF Information: Double-checking master pkgdef file
12.05.2017 19:20:44 - PKGDEF Information: PkgDefManagement initialized
12.05.2017 19:20:44 - PKGDEF Information: RootFolder, Path: C:\Program Files (x86)\Visual Studio\2017\Community\

12.05.2017 19:20:44 - PKGDEF Information: ShellFolder, Path: C:\Program Files (x86)\Visual Studio\2017\Community\

12.05.2017 19:20:44 - PKGDEF Information: PkgDefSearchPath, Path: C:\Program Files (x86)\Visual Studio\2017\Community\Common7\IDE\Extensions;C:\Program Files (x86)\Visual Studio\2017\Community\Common7\IDE\CommonExtensions;C:\Program Files (x86)\Visual Studio\2017\Community\Common7\IDE\devenv.admin.pkgdef

12.05.2017 19:20:44 - PKGDEF Information: ImageManifestSearchPath, Path: C:\Program Files (x86)\Visual Studio\2017\Community\Common7\IDE\Extensions;C:\Program Files (x86)\Visual Studio\2017\Community\Common7\IDE\CommonExtensions;C:\Users\Denis\AppData\Local\Microsoft\VisualStudio\15.0_69dcc73d\Extensions

12.05.2017 19:20:44 - PKGDEF Information: ApplicationExtensionsFolder, Path: C:\Program Files (x86)\Visual Studio\2017\Community\Common7\IDE\Extensions

12.05.2017 19:20:44 - PKGDEF Information: "ZeroImpact" = dword:0, "MergeRegistry" = dword:3
12.05.2017 19:20:44 - PKGDEF Information: Could not find ConfigurationChanged timestamp., PKGDEF: 80070002
12.05.2017 19:20:44 - PKGDEF Information: User extensions enabled by setting, Path: HKEY_CURRENT_USER\Software\Microsoft\VisualStudio\15.0_69dcc73d\ExtensionManager\EnableAdminExtensions

12.05.2017 19:20:44 - PKGDEF Information: PkgDefCache flags, PKGDEF: 7001
12.05.2017 19:20:44 - PKGDEF Information: Could not find ConfigurationChanged timestamp., PKGDEF: 80070002
12.05.2017 19:20:44 - PKGDEF Information: PkgDefManagement startup complete
12.05.2017 19:20:45 - The extension with ID '40c1d2b5-528b-4966-a7b1-1974e3568abe' is not installed to Visual Studio Community 2017.
12.05.2017 19:20:51 - The following target products have been selected...
12.05.2017 19:20:51 - 	Visual Studio Community 2017
12.05.2017 19:20:51 - 
12.05.2017 19:20:51 - Beginning to install extension to Visual Studio Community 2017...
12.05.2017 19:20:51 - Install Error : Microsoft.VisualStudio.ExtensionManager.InstallByMsiException: The InstalledByMSI element in extension Visual Rust cannot be 'true' when installing an extension through the Extensions and Updates Installer.  The element can only be 'true' when an MSI lays down the extension manifest file.
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.CheckForInstallBlockers(InstallableExtensionImpl extension, IInstalledExtensionList modifiedInstalledExtensionsList, Boolean isNestedExtension, InstallFlags flags, Boolean& performExtensionUpdate)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.InstallInternal(InstallableExtensionImpl extension, InstallFlags installFlags, IDictionary`2 extensionsInstalledSoFar, List`1 extensionsUninstalledSoFar, IInstalledExtensionList modifiedInstalledExtensionsList, AsyncOperation asyncOp, IProgress`1 progress)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.BeginInstall(IInstallableExtension installableExtension, InstallFlags installFlags, AsyncOperation asyncOp)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.InstallWorker(IInstallableExtension extension, InstallFlags installFlags, AsyncOperation asyncOp)
```

**Comment by mateusmedeiros** [comment]
I'm on Creators Update (15063) and it happens to me as well.

**Comment by Skyne98** [comment]
As well, Creators Update, VS2017 without Build Tools

**Comment by bbi-ashmatheson** [comment]
This occurs for me on both VS 2015 and 2017. Same log files.

**Comment by tchabada** [comment]
Is there some previous version build which works for 20017 please?

**Comment by ioncodes** [comment]
Same on my side, VS2017 + Creators Update.
I will look into this part: Microsoft.VisualStudio.ExtensionManager.InstallByMsiException: The InstalledByMSI element in extension Visual Rust cannot be 'true' when installing an extension through the Extensions and Updates Installer.  The element can only be 'true' when an MSI lays down the extension manifest file.

**Comment by ioncodes** [comment]
It appears to be some sort of misconfiguration in the manifest file of the vsix. Maybe [this](https://social.msdn.microsoft.com/Forums/vstudio/en-US/12faa07b-af39-43f0-a62a-e83d2e0983c7/how-to-avoid-the-installedbymsi-element-in-extension-cannot-be-true-error?forum=vsx) helps you.

**Comment by ioncodes** [comment]
I was able to bypass the error, [here](https://gist.github.com/ioncodes/43055423dba7590f2cbadf3381e79949) is how. I will make a new Issue with this, and post there the fixed files.

**Comment by ioncodes** [comment]
https://github.com/PistonDevelopers/VisualRust/issues/281

**Comment by indygreg** [comment]
My fork contains a hacky commit to allow building a working installer for VS2017. I don't have plans to submit a pull request because I'm not sure what how appropriate the changes are. I just wanted to get VisualRust installed in VS2017 for some weekend hacking :)

**Comment by mateusmedeiros** [comment]
Last build after #282 installed just fine here. 👍

**Comment by denis-golubev** [comment]
Worked well for me, too 👍
